### PR TITLE
Traffic Ops Rewrite: /api/1.2/servers endpoint

### PR DESCRIPTION
**Pull request by dewrich**
This is an example endpoint WITHOUT sqlx

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/243/

File: traffic_ops/traffic_ops_golang/servers.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+		var routerPortName sql.NullString
+		var status sql.NullString
+		var statusId sql.NullInt64
+		var tcpPort sql.NullInt64
+		var serverType sql.NullString
+		var serverTypeId sql.NullInt64
+		var updPending sql.NullBool
+		var xmppId sql.NullString
+		var xmppPasswd sql.NullString
+		if err := rows.Scan(&cachegroup, &cachegroupId, &cdnId, &cdnName, &domainName, &guid, &hostName, &httpsPort, &id, &iloIpAddress, &iloIpGateway, &iloIpNetmask, &iloPassword, &iloUsername, &interfaceMtu, &interfaceName, &ip6Address, &ip6Gateway, &ipAddress, &ipGateway, &ipNetmask, &lastUpdated, &mgmtIpAddress, &mgmtIpGateway, &mgmtIpNetmask, &offlineReason, &physLocation, &physLocationId, &profile, &profileDesc, &profileId, &rack, &revalPending, &routerHostName, &routerPortName, &status, &statusId, &tcpPort, &serverType, &serverTypeId, &updPending, &xmppId, &xmppPasswd); err != nil {
```
Comment (alficles) [inline traffic_ops/traffic_ops_golang/servers.go]: Why not just scan directly into the new object? This way is useful if you've got nullable columns that need to be handled carefully, but you've already taken care of that with `coalesce`.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/200/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/201/

**Pull request closed without merging**